lamagraph-core/src/Core/MemoryManager.hs

```diff
@@ -0,0 +1,383 @@
+              Just (info, _) -> Just (constructNewInfo info, localFlag)
+          )
+          addressNum
+       where
+        newPort = Port (Just $ ActualAddress (localNodeWithAddress ^. originalAddress)) portId
+        constructNewInfo info@(NodePortInfo secPortsInfo _) = case connectedToPortId of
+          Primary -> set primeP (Just newPort) info
+          Id i -> set secP (replace i (Just newPort) secPortsInfo) info
+
+updatePortsInfoByEdge ::
+  forall portsNumber maxNumOfChangedNodes.
+  (KnownNat portsNumber, KnownNat maxNumOfChangedNodes) =>
+  Map maxNumOfChangedNodes (NodePortsInfo portsNumber, LocalFlag) ->
+  Edge portsNumber ->
+  Map maxNumOfChangedNodes (NodePortsInfo portsNumber, LocalFlag)
+updatePortsInfoByEdge oldInfoVec (Edge leftE rightE) =
+  update
+    rightAddrNum
+    leftAddrNum
+    (rightE ^. idOfPort)
+    (leftE ^. idOfPort)
+    $ update leftAddrNum rightAddrNum (leftE ^. idOfPort) (rightE ^. idOfPort) oldInfoVec
+ where
+  leftAddrNum = leftE ^. addressOfVertex
+  rightAddrNum = rightE ^. addressOfVertex
+  update addrToUpdate addrToWhichUpdate connectedToPortId portId infoVec =
+    insertWith
+      infoVec
+      ( \case
+          Nothing -> Just (constructNewInfo (NodePortInfo def Nothing), External)
+          Just (info, _) -> Just (constructNewInfo info, External)
+      )
+      addrToUpdate
+   where
+    newPort = Port (Just $ ActualAddress addrToWhichUpdate) portId
+    constructNewInfo info@(NodePortInfo secPortsInfo _) = case connectedToPortId of
+      Primary -> set primeP (Just newPort) info
+      Id i -> set secP (replace i (Just newPort) secPortsInfo) info
+
+-- Check if `Node` is active
+newNodeIsActive ::
+  (KnownNat portsNumber) => Node portsNumber -> Bool
+newNodeIsActive node =
+  case node ^. primaryPort . portConnectedToId of
+    Primary -> True
+    _ -> False
+
+updateMM ::
+  forall (cellsNumber :: Nat) (portsNumber :: Nat) (edgeNumber :: Nat) (dom :: Domain) (maxNumOfChangedNodes :: Nat).
+  ( KnownNat cellsNumber
+  , KnownNat portsNumber
+  , KnownNat edgeNumber
+  , 1 <= cellsNumber
+  , CLog 2 cellsNumber ~ 16
+  , 1 <= edgeNumber
+  , CLog 2 edgeNumber ~ 16
+  , KnownNat maxNumOfChangedNodes
+  ) =>
+  ( Signal dom ActualAddressNumber ->
+    Signal dom (Maybe (ActualAddressNumber, Maybe (Node portsNumber))) ->
+    Signal dom (Maybe (Node portsNumber))
+  ) ->
+  Signal dom (MemoryManager cellsNumber) ->
+  Signal dom (Delta cellsNumber edgeNumber portsNumber) ->
+  ( Signal dom (MemoryManager cellsNumber)
+  , Signal dom ActualAddressNumber ->
+    Signal dom (Maybe (ActualAddressNumber, Maybe (Node portsNumber))) ->
+    Signal dom (Maybe (Node portsNumber))
+  )
+-- it is possible to merge all foldl into one
+updateMM ram memoryManager delta = (MemoryManager <$> markedBusyBitMap <*> newActives, ram)
+ where
+  localNodesSignal = unbundle $ (^. newNodes) <$> delta :: _ (Signal _ _)
+  activePairSignal = (^. activePair) <$> delta
```

Comment:
This is equivalent to 
```suggestion
  activePairSignal = view activePair <$> delta
```
and IMO `view` is much more readable than partial application of an infix operator. 
I'd suggest changing this pattern to `view` across all files